pkg/genevalogging/genevalogging.go

```diff
@@ -548,4 +566,223 @@ func (g *genevaLogging) CreateOrUpdate(ctx context.Context) error {
 			},
 		},
 	})
+
+	if err != nil {
+		return err
+	}
+
+	return g.applyDaemonSet(&appsv1.DaemonSet{
```

Comment:
there's a lot of repetition here.  Can you define a separate function which builds 99% of your DaemonSet, then customise it as required?

Reply:
Reduced back to one DS.